go-bindings/Makefile

```
@@ -1,29 +1,50 @@
-SRC_DIR=$(PWD)/../src
-BUILD_DIR=$(PWD)/../build
+export CGO_ENABLED := 1
+GO=go
 
-GO="go"
-COVERAGE_OUTPUT ?= coverage.out
-
-.PHONY: default vet test clean
+OS := $(shell uname -s)
+ARCH := $(shell uname -m)
 
-default: prepare vet test clean
+COVERAGE_OUTPUT ?= coverage.out
 
 MAKEFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
+
+BUILD_DIR=$(CURDIR)/../build
 CURR_DIR := $(dir $(MAKEFILE_PATH))
+SRC_DIR=$(CURDIR)/../src
 
-CGO_ENABLED := 1
+GMP_PREFIX := /usr/local
 
-CGO_LDFLAGS ?= "\
--L$(CURR_DIR)../build/depends/mimalloc \
--L$(CURR_DIR)../build/depends/relic/lib \
--L$(CURR_DIR)../build/src \
--ldashbls -lrelic_s -lmimalloc-secure -lgmp"
+ifeq ("$(OS)", "Darwin")
+  ifneq ($(wildcard /opt/local/bin/port),)
+    GMP_PREFIX := /opt/local
+  endif
+  ifeq ("$(ARCH)", "arm64")
+    ifneq ($(wildcard /opt/homebrew/bin/brew),)
+      GMP_PREFIX := /opt/homebrew
+    endif
+  endif
+else ifeq ("$(OS)", "Linux")
+  ifneq ($(wildcard /home/linuxbrew/.linuxbrew/bin/brew),)
+    GMP_PREFIX := /home/linuxbrew/.linuxbrew
+  endif
+endif
 
 CGO_CXXFLAGS ?= "\
 -I$(CURR_DIR)../build/depends/relic/include \
 -I$(CURR_DIR)../depends/mimalloc/include \
 -I$(CURR_DIR)../depends/relic/include \
--I$(CURR_DIR)../include"
+-I$(CURR_DIR)../include \
+-I$(GMP_PREFIX)/include"
+
+CGO_LDFLAGS ?= "\
+-L$(CURR_DIR)../build/depends/mimalloc -lmimalloc-secure \
+-L$(CURR_DIR)../build/depends/relic/lib -lrelic_s \
+-L$(CURR_DIR)../build/src -ldashbls \
+-L$(GMP_PREFIX)/lib -lgmp"
```

Wrong linking order?
```suggestion
-L$(CURR_DIR)../build/src -ldashbls \
-L$(CURR_DIR)../build/depends/mimalloc -lmimalloc-secure \
-L$(CURR_DIR)../build/depends/relic/lib -lrelic_s \
-L$(GMP_PREFIX)/lib -lgmp"
```

---

Resolved in latest push.